### PR TITLE
feat: curve staking

### PR DESCRIPTION
**Pull request by jpsains**

### CODE REVIEW

File: src/adapters/curve/products/staking/curveStakingAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  ): Promise<Erc20Metadata[]> {
+    const { underlyingTokens } = await this.fetchPoolMetadata(
+      protocolTokenAddress,
+    )
+
+    return underlyingTokens
+  }
+
+  private async fetchPoolMetadata(protocolTokenAddress: string) {
+    console.log({ protocolTokenAddress })
```
Comment (jpsains) [inline src/adapters/curve/products/staking/curveStakingAdapter.ts:303]: ill remove

File: src/adapters/curve/common/getPoolData.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import { ZERO_ADDRESS } from '../../../core/constants/ZERO_ADDRESS'
+import { CustomJsonRpcProvider } from '../../../core/utils/customJsonRpcProvider'
+import { getTokenMetadata } from '../../../core/utils/getTokenMetadata'
+import { Chain } from '../../../defiProvider'
+import { Erc20Metadata } from '../../../types/erc20Metadata'
+import { MetaRegistry__factory } from '../contracts'
+import { CURVE_META_REGISTRY_CONTRACT } from '../products/pool/curvePoolAdapter'
+
+const LOW_LP_TOKEN_SUPPLY = 1
+export async function getPoolData(
```
Comment (bergarces) [inline src/adapters/curve/common/getPoolData.ts:12]: Shouldn't this method replace `getPoolData` in `CurvePoolAdapter`?

File: src/adapters/curve/products/staking/curveStakingAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+  poolAdapter: IProtocolAdapter & IMetadataBuilder
+
+  constructor(params: ProtocolAdapterParams) {
+    super(params)
+
+    this.poolAdapter = params.adaptersController.fetchAdapter(
+      params.chainId,
+      params.protocolId,
+      'pool',
```
Comment (bergarces) [inline src/adapters/curve/products/staking/curveStakingAdapter.ts:57]: Should be `this.productId`
```suggestion
      this.productId,
```

File: src/core/adapters/SimplePoolAdapter.ts [hunk 4]
```diff
@@ -136,14 +141,19 @@ export abstract class SimplePoolAdapter implements IProtocolAdapter {
...
+      ),
       underlyingTokens: await this.fetchUnderlyingTokensMetadata(
         protocolTokenAddress,
       ),
-      fromBlock,
-      toBlock,
-      from: userAddress,
-      to: undefined,
+      filter: {
+        smartContractAddress: protocolTokenAddress,
```
Comment (bergarces) [inline src/core/adapters/SimplePoolAdapter.ts:151]: I like this solution for now, but we need to keep an eye if there's a pattern for other staking adapters in which the filter address is different than the protocol token address, as then we might want to create a new template adapter.

**Review by bergarces — APPROVED** [review]

**Pull request merged**